lib/pages/login_page.dart

```diff
@@ -94,24 +92,27 @@ class _LoginPageState extends State<LoginPage> {
             )));
   }
 
-  Future signIn(BuildContext) async {
+  Future signIn() async {
     showLoadingSpinner();
     final authenticationService = Provider.of<AuthenticationService>(context, listen: false);
 
-    var userUuid =
-        await authenticationService.login(emailController.text, passwordController.text);
+    var userUuid = await authenticationService.login(emailController.text, passwordController.text);
     if (userUuid != null) {
       widget.logger.i('starting to fetch user data');
       var userData = await authenticationService.fetchUserData(userUuid);
       if (userData == null) {
         widget.logger.e('User data was null');
         return;
       }
-      Provider.of<UserProvider>(context, listen: false).setUser(userData);
-      Navigator.of(context).popUntil(ModalRoute.withName(HomePage.route));
+      if (context.mounted) {
+        Provider.of<UserProvider>(context, listen: false).setUser(userData);
+        Navigator.of(context).popUntil(ModalRoute.withName(HomePage.route));
+      }
     } else {
       showToast('Wrong email or password.');
-      Navigator.pop(context);
+      if (context.mounted) {
```

Comment:
nie wiedzialem ze trzeba cokolwiek sprawdzac przy uzywaniu contextu. w moim kursie pominęli :/

Reply:
Android studio ma kijowy linter że wam nie mówi xdd

Reply:
Nie, Android Studio tez sie o to czepia: "Don't use 'BuildContext's across async gaps. (Documentation)  Try rewriting the code to not reference the 'BuildContext" 😅